porsche-design-system/porsche-design-system
# Upgrade UXPin Merge CLI and set up styles to inject

### Pull Request Checklist




- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### Scope

This PR upgrades UXPin Merge CLI to the latest version (`3.3.0`) to take advantage of a feature I added especially for Porsche: now  users can inject any custom tag (`<style>` or `<script>`) in UXPin Editor and Preview pages, using a new option called `pageHeadTags` in the UXPin config file. It accepts either a string or an array of strings if multiple tags need to be injected.

Details in the PR: https://github.com/UXPin/uxpin-merge-tools/pull/401

Additional note: since `3.2.0`, Merge CLI is much faster to serialize TS components (it used to take more than 500 ms for every single component), so the upgrade is highly recommended 🚄  😄 


#### Resolved Issue

Fix #2427 

#### How to test

- Run UXPin in experimental mode: `yarn start` from `packages/components-react/projects/uxpin-wrapper`
- Add a TextField component to the canvas
- Set the `type` field to `"search"`
- Launch UXPin Preview
- Enter some content: check that the ugly X icon provided by the browser's default styles is no more displayed, in Chrome. Only the nice one coming from the Porsche component should be displayed.

In Editor:
 
![image](https://github.com/porsche-design-system/porsche-design-system/assets/5546996/8f888994-4c9a-476b-a31d-a3cf184d9ed2)

In Preview:

![image](https://github.com/porsche-design-system/porsche-design-system/assets/5546996/9c6745dd-301b-4711-982c-8671d0f4a4af)
